### PR TITLE
[FIX](map)fix column map for offset next_array_item_rowid order

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.84 seconds
 stream load tsv:          555 seconds loaded 74807831229 Bytes, about 128 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.9 seconds inserted 10000000 Rows, about 334K ops/s
 storage size: 17162081593 Bytes

File: be/src/olap/rowset/segment_v2/column_writer.cpp [hunk 1]
```diff
@@ -1083,17 +1083,19 @@ Status MapColumnWriter::append_data(const uint8_t** ptr, size_t num_rows) {
...
+            auto data = *(data_ptr + 2 + i);
+            auto nested_null_map = *(data_ptr + 2 + 2 + i);
+            RETURN_IF_ERROR(
+                    _kv_writers[i]->append(reinterpret_cast<const uint8_t*>(nested_null_map),
+                                           reinterpret_cast<const void*>(data), element_cnt));
+        }
     }
+    // make sure the order : offset writer flush next_array_item_ordinal after kv_writers append_data
+    // because we use _kv_writers[0]->get_next_rowid() to set next_array_item_ordinal in offset page footer
+    RETURN_IF_ERROR(_offsets_writer->append_data(&offsets_ptr, num_rows));
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/column_writer.cpp:1098]: there is the same problem for array

**Review by xiaokang — COMMENTED** [review]
add regression test

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by vinlee19 — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run clickbench

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.41 seconds
 stream load tsv:          540 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.6 seconds inserted 10000000 Rows, about 337K ops/s
 storage size: 17161982913 Bytes

**Comment by shuke987** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.63 seconds
 stream load tsv:          557 seconds loaded 74807831229 Bytes, about 128 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162070144 Bytes

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**